pkg/tfshim/sdk-v2/resource_config.go

```diff
@@ -27,3 +34,24 @@ func (c v2ResourceConfig) IsSet(key string) bool {
 
 	return false
 }
+
+func (c v2ResourceConfig) GetRawConfigMapWithUnknown() (map[string]any, error) {
+	jsonConfigMap := map[string]any{}
+	ctyValue := c.tf.CtyValue
+	if !ctyValue.IsWhollyKnown() {
+		msg := fmt.Sprintf("It looks like you're trying to use the provider's terraformConfig function. " +
```

Comment:
This error message would be better placed in the terraformConfig function handler that is defining how it operates. As written it bakes in the assumption that only the terraformConfig function handler may call GetRawConfigMapWithUnknown (btw I agree with the above comment would like to request a better name here).